src/server/scripts/Northrend/zone_borean_tundra.cpp

```diff
@@ -34,6 +34,141 @@
 #include "TemporarySummon.h"
 #include "WorldSession.h"
 
+ /*######
+ ## Quest 11590: Abduction
+ ######*/
+
+// NPC 25316: Beryl Sorcerer
+enum BerylSorcerer
+{
+    EVENT_FROSTBOLT                                = 1,
+    EVENT_ARCANE_CHAINS,
+    EVENT_DESPAWN,
+
+    NPC_LIBRARIAN_DONATHAN                         = 25262,
+    NPC_CAPTURED_BERLY_SORCERER                    = 25474,
+
+    SPELL_FROSTBOLT                                = 9672,
+    SPELL_ARCANE_CHAINS                            = 45611,
+    SPELL_ARCANE_CHAINS_CHARACTER_FORCE_CAST       = 45625,
+    SPELL_ARCANE_CHAINS_SUMMON_CHAINED_MAGE_HUNTER = 45626,
+    SPELL_COSMETIC_ENSLAVE_CHAINS_SELF             = 45631,
+    SPELL_ARCANE_CHAINS_CHANNEL_II                 = 45735
+};
+
+struct npc_beryl_sorcerer : public CreatureAI
+{
+    npc_beryl_sorcerer(Creature* creature) : CreatureAI(creature), _chainsCast(false) { }
+
+    void JustAppeared() override
+    {
+        me->SetReactState(REACT_AGGRESSIVE);
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        if (me->IsValidAttackTarget(who))
+            AttackStart(who);
+
+        _events.ScheduleEvent(EVENT_FROSTBOLT, 3s, 4s);
+    }
+
+    void SpellHit(WorldObject* caster, SpellInfo const* spellInfo) override
+    {
+        if (_chainsCast)
+            return;
+
+        if (spellInfo->Id == SPELL_ARCANE_CHAINS)
+            if (Player* player = caster->ToPlayer())
+            {
+                _playerGUID = player->GetGUID();
+                _chainsCast = true;
```

Comment:
is this supposed to be set to false somewhere again ? like on Reset() ?

Reply:
He despawns in script

Reply:
false is set on spawn